static/js/components/NewApplication.js

```diff
@@ -0,0 +1,146 @@
+/* global SETTINGS: false */
+import React from "react"
+import { compose } from "redux"
+import { connect } from "react-redux"
+import { mutateAsync } from "redux-query"
+import { connectRequest } from "redux-query-react"
+import { createStructuredSelector } from "reselect"
+import {
+  Dropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem
+} from "reactstrap"
+import { partial } from "ramda"
+
+import { setDrawerOpen } from "../reducers/drawer"
+import users, { currentUserSelector } from "../lib/queries/users"
+import bootcamps, { bootcampRunsSelector } from "../lib/queries/bootcamps"
+import applications, { createAppQueryKey } from "../lib/queries/applications"
+import { isQueryPending } from "../lib/queries/util"
+
+import type { User } from "../flow/authTypes"
+import type { BootcampRun } from "../flow/bootcampTypes"
+
+type Props = {
+  currentUser: User,
+  bootcampRuns: ?Array<BootcampRun>,
+  appliedRunIds: Array<number>,
+  createAppIsPending: boolean,
+  createNewApplication: (bootcampRunId: number) => void,
+  setDrawerOpen: (newState: boolean) => void
+}
+
+type State = {
+  selectedBootcamp: ?BootcampRun,
+  dropdownOpen: boolean
+}
+
+const INITIAL_STATE: State = {
+  selectedBootcamp: null,
+  dropdownOpen:     false
+}
+
+export class NewApplication extends React.Component<Props, State> {
```

Comment:
I'm just learning about react hooks and avoiding class components now. Looks promising. I'd like to keep this one as-is since it works, and for future components I'll try to get a better handle on that approach. This might also be something we want to discuss as a team and build some patterns around. It seems like we can end up building some components that are difficult to understand if we don't have some general agreed-upon guidelines